packages/dma-library/src/strategies/aave-like/multiply/close/types.ts

```diff
@@ -1,16 +1,28 @@
+import { FlashloanProvider } from '@dma-library/types'
 import * as Strategies from '@dma-library/types/strategies'
 import * as StrategyParams from '@dma-library/types/strategy-params'
-import { WithFlashloanToken } from '@dma-library/types/strategy-params'
+import { WithFlashLoanArgs } from '@dma-library/types/strategy-params'
+import { BigNumber } from 'bignumber.js'
 
 export type AaveLikeCloseArgs = StrategyParams.WithAaveLikeMultiplyStrategyArgs &
-  StrategyParams.WithCloseToCollateralFlag
+  StrategyParams.WithCloseToCollateralFlag &
+  Partial<StrategyParams.WithFlashLoanArgs>
 
 export type AaveLikeExpandedCloseArgs = AaveLikeCloseArgs &
   StrategyParams.WithProtocolData & {
     collateralToken: AaveLikeCloseArgs['collateralToken'] & { address: string }
     debtToken: AaveLikeCloseArgs['debtToken'] & { address: string }
-    flashloanToken: WithFlashloanToken['flashloanToken'] & { address: string }
+  } & WithFlashLoanArgs
+
+export type CloseFlashloanArgs = {
+  token: {
```

Comment:
I think we could create a `Token` type being:
```
export type Token : {
  address: string
  precision: number
  symbol: string
}
```

and then a `TokenAmount` type being:
```
export type TokenAmount:  {
  token: Token
  amount: BigNumber
}
```

To try to start having a bit more semantics in the code

Reply:
Yes it's a good idea!